File: backend/src/TaskManager.Api/Dtos/TaskDto.cs
namespace TaskManager.Api.Dtos;

/// <summary>
/// Forma exacta del JSON que devuelve la API al cliente.
/// Renombramos StatusCode → Status / PriorityCode → Priority para una shape más limpia.
/// </summary>
public record TaskDto(
    long Id,
    string Title,
    string? Description,
    string Status,
    string StatusLabel,
    string Priority,
    string PriorityLabel
);

File: mobile/src/features/tasks/data/TaskDto.ts
import type { TaskPriority, TaskStatus } from '../domain/Task';

/**
 * Forma exacta del JSON que retorna la API.
 * No usar fuera de la capa data — el resto de la app consume `Task`.
 */
export interface TaskDto {
  id: number;
  title: string;
  description: string | null;
  status: TaskStatus;
  statusLabel: string;
  priority: TaskPriority;
  priorityLabel: string;
}
